src/client/theme-default/styles/components/vp-doc.css

```diff
@@ -310,23 +310,23 @@
 }
 
 .vp-doc [class*='language-'] code {
-  display: flex;
-  width: max-content;
+  display: block;
+  padding: 0 24px;
+  width: fit-content;
   min-width: 100%;
-  flex-direction: column;
   line-height: var(--vp-code-line-height);
   font-size: var(--vp-code-font-size);
   color: var(--vp-code-block-color);
   transition: color 0.5s;
 }
 
-.vp-doc [class*='language-'] code .line {
-  padding: 0 24px;
-}
-
 .vp-doc [class*='language-'] code .highlighted {
   background-color: var(--vp-code-line-highlight-color);
   transition: background-color 0.5s;
+  margin: 0 -24px;
```

Comment:
Here, I prefer applying these layout properties on `.line` instead of `.highlighted` to keep the layout of each line consistent. And the `display` can be `block` since it actually fills the whole line, not an inline element anymore. WDYT?

Reply:
Putting it on .line directly was causing some issue like extra blank line at end IG. Regarding that block thing, I'll check it.

Reply:
I see. Then I think so far `inline-block` is the best choice, whatever it's on `.line` or `.highighted`. No problem from my side. 👍

Reply:
weird...

Take the `[[toc]]` for example. The generated code has 2 lines actually, which causes the extra blank line as you mentioned.

![image](https://user-images.githubusercontent.com/206848/184654946-4a248a6e-ab85-4fea-ac7b-adf6f945dc7f.png)

Anyway, let's take this.

Reply:
Yeah we had discussed this stuff internally too before alpha. Then it was happening only on Firefox. Ideally, that last line should not be there IMO. I'll check if we can fix that from the plugin itself.

Reply:
Okay, so removing that extra line has not done simplified anything. Because it's actually not guaranteed to have a `.line` or even a span inside `code`. Gonna go ahead with these changes themselves.